RawMediaAccess/app/src/main/java/com/microsoft/acs/calling/rawmediaaccess/FrameGenerator.java

```diff
@@ -0,0 +1,123 @@
+package com.microsoft.acs.calling.rawmediaaccess;
+
+import android.util.Log;
+
+import com.azure.android.communication.calling.SoftwareBasedVideoFrameSender;
+import com.azure.android.communication.calling.VideoFormat;
+import com.azure.android.communication.calling.VideoFrameSender;
+import com.azure.android.communication.calling.VideoFrameSenderChangedEvent;
+import com.azure.android.communication.calling.VideoFrameSenderChangedListener;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+
+public class FrameGenerator implements VideoFrameSenderChangedListener {
+
+    private VideoFrameSender videoFrameSender;
+    private Thread frameIteratorThread;
+    private final Random random;
+    private volatile boolean stopFrameIterator = false;
+
+    public FrameGenerator() {
+
+        random = new Random();
+    }
+
+    public void FrameIterator() {
+
+        ByteBuffer plane = null;
+        while (!stopFrameIterator && videoFrameSender != null) {
+
+            plane = GenerateFrame(plane);
+        }
+
+        Log.d("FrameGenerator", "FrameGenerator.FrameIterator trace, 1st looper stopped");
+    }
+
+    private ByteBuffer GenerateFrame(ByteBuffer plane) {
+
+        try {
+
+            VideoFormat videoFormat = videoFrameSender.getVideoFormat();
+            if (plane == null || videoFormat.getStride1() * videoFormat.getHeight() != plane.capacity()) {
+
+                plane = ByteBuffer.allocateDirect(videoFormat.getStride1() * videoFormat.getHeight());
+                plane.order(ByteOrder.nativeOrder());
+            }
+
+            int bandsCount = random.nextInt(15) + 1;
+            int bandBegin = 0;
+            int bandThickness = videoFormat.getHeight() * videoFormat.getStride1() / bandsCount;
+
+            for (int i = 0; i < bandsCount; ++i) {
+
+                byte greyValue = (byte) random.nextInt(254);
+                java.util.Arrays.fill(plane.array(), bandBegin, bandBegin + bandThickness, greyValue);
+                bandBegin += bandThickness;
+            }
+
+            SoftwareBasedVideoFrameSender sender = (SoftwareBasedVideoFrameSender) videoFrameSender;
```

Comment:
Not required, but I remember you had a version that supported HardwareBasedVideoFrameSender as well, right? I think it would be nice for customers to include on the quickstart if that's easy.

Reply:
for now is not working so we are not including that for now